src/wpas/wpa_controller.c

```diff
@@ -820,13 +820,15 @@ __wpa_controller_send_commandf(struct wpa_controller *ctrl, const char *name, ch
 int
 wpa_controller_qrcode(struct wpa_controller *ctrl, const char *dpp_uri, uint32_t *bootstrap_id)
 {
-    char reply[WPA_MAX_MSG_SIZE];
-    size_t reply_length = sizeof reply;
+    char reply[WPA_MAX_MSG_SIZE + 1];
+    size_t reply_length = WPA_MAX_MSG_SIZE;
```

Comment:
`wpa_controller_send_commandf` requires this, on input, to describe the full length of the buffer. So this should still be `sizeof reply` as before. The implementation of `wpa_controller_send_commandf` can then terminate the buffer and allow the max size, since you've now increased it by 1.